src/linux/ebpf.cpp

```diff
@@ -96,4 +98,51 @@ Try<int> load(const Program& program)
   return *fd;
 }
 
+namespace cgroups2 {
+
+Try<Nothing> attach(int fd, const string& cgroup)
+{
+  Try<int> cgroupFd = os::open(cgroup, O_DIRECTORY | O_RDONLY);
+  if (cgroupFd.isError()) {
+    return Error(
+        "Failed to find the file descriptor of '" + cgroup +
+        "': " + cgroupFd.error());
+  }
+
+  bpf_attr attr;
+  memset(&attr, 0, sizeof(attr));
+  attr.attach_type = BPF_CGROUP_DEVICE;
+  attr.target_fd = *cgroupFd;
+  attr.attach_bpf_fd = fd;
+  // BPF_F_ALLOW_MULTI allows multiple eBPF programs to be attached to a single
+  // cgroup and determines the order in which attached programs will run.
```

Comment:
we don't want multiple programs within a single cgroup, rather we want to replace them, we can add a TODO to perform replacement for now